mockk/src/main/kotlin/io/mockk/impl/instantiation/JvmMockFactoryHelper.kt

```diff
@@ -67,7 +67,7 @@ internal object JvmMockFactoryHelper {
                 .filter { it.kind != KParameter.Kind.INSTANCE }
                 .indexOfFirst { it.isVararg }
         else
-            parameters.indexOfFirst { it.isVarArgs }
+            if (isVarArgs) parameterTypes.size - 1 else -1
```

Comment:
I have doubts that this is semantically same expressions. You should note that Kotlin varargs can go any place in parameters! Can you please check?

Reply:
Indeed that's so for Kotlin, but we're using Java's `Method` here. I've moved this method out of `JvmMockFactoryHelper ` and introduced a simple test for it.